### DISCUSSION TITLE
Bulk issuance error

### DISCUSSION THREAD

**Discussion by gohilamariappan** [Sunbird-RC/community#1038] [Issues] [answered]
HI Team!

we are getting an error while deploying bulk issuance.  

@holashchand Please suggest..
############
Failed to find configuration ./application-default.yml
time="2024-06-25T11:34:15Z" level=info msg="Public key url : http://keycloak:8080/auth/realms/sunbird-rc"
time="2024-06-25T11:34:15Z" level=error msg="Error while fetching public key from keycloakGet \"http://keycloak:8080/auth/realms/sunbird-rc\": dial tcp: lookup keycloak on 10.3.0.10:53: no such host"
time="2024-06-25T11:34:15Z" level=info msg="Using the public key -----BEGIN PUBLIC KEY-----\n\n-----END PUBLIC KEY-----\n"
time="2024-06-25T11:34:15Z" level=error msg="Error parsing public key from keycloak : []%!(EXTRA asn1.SyntaxError=asn1: syntax error: sequence truncated)
##########

We referred the below compose file for deployment.
https://github.com/Sunbird-RC/sunbird-rc-core/blob/main/docker-compose.yml#L442

Regards,
Gohila

**Comment by holashchand** [comment, marked as answer]
Hi @gohilamariappan 
The bulk_issuance service is moved [here](https://github.com/Sunbird-RC/auxillary-services/tree/main/bulk_issuance). The required files can be found here

**Comment by gohilamariappan** [comment]
@holashchand  The files are already present in the container. but its not picking at the time of starting.
Please suggest.

################
/app/bulk_issuance # ls
Dockerfile     bulk_issuance  db             go.sum         main.go        services       utils
Makefile       config         go.mod         interfaces     pkg            swagger_gen
/app/bulk_issuance # cd config/
/app/bulk_issuance/config # ls
application-default.yml  config.go
/app/bulk_issuance/config # vi application-default.yml
/app/bulk_issuance/config #
#########

Regards,
Gohila

**Reply by holashchand** [reply]
Hi @gohilamariappan 
The file is present in config directory but it's trying to look into the working directory, we can do two thing -
1. COPY file in the working directory and build again
2. add file as volume to the bulk issuance service `file-location:/app/bulk_issuance/application-default.yml`

**Comment by gohilamariappan** [comment]
@holashchand Tried above solutions. facing issues.
1. COPY file in the working directory and build again
######
 > [8/9] RUN go test -v ./...:
1.218 found packages config (config.go) and main (main.go) in /app/bulk_issuance
------
Dockerfile:8
--------------------
   6 |     RUN go mod download
   7 |     RUN make deps
   8 | >>> RUN go test -v ./...
   9 |     RUN GOFLAGS=" -tags=musl" SPEC_FILE="./interfaces/bulk_issuance-api.yaml" make all
  10 |     EXPOSE 5665
--------------------
ERROR: failed to solve: process "/bin/sh -c go test -v ./..." did not complete successfully: exit code: 1
########

2.  add file as volume to the bulk issuance service file-location:/app/bulk_issuance/application-default.yml

we are using kubernetes , tried attaching as a configmap in the location, that time other files not available , so container is unable to start.

Kindly suggest.

Regards,
Gohila

**Reply by holashchand** [reply]
Hi @gohilamariappan , you can use cmd `make docker` to build the docker image using method 1

**Comment by PrasadAthani** [comment]
Hi Team 
Here i am attaching logs of Bulk Issuance service 
`sudo docker logs -f 48378fbbcbc8
Failed to find configuration ./application-default.yml
time="2024-07-03T09:21:11Z" level=info msg="Public key url : http://keycloak:8080/auth/realms/sunbird-rc"
time="2024-07-03T09:21:11Z" level=info msg="Got response {\"realm\":\"sunbird-rc\",\"public_key\":\"MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAuC65RhunHw/V-connect\",\"account-service\":\"http://localhost:8080/auth/realms/sunbird-rc/account\",\"tokens-not-before\":1607576887}"
time="2024-07-03T09:21:11Z" level=info msg="Using the public key -----BEGIN PUBLIC KEY-----\nMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMII
time="2024-07-03T09:21:11Z" level=info msg="Using db host=db port=5432 user=postgres password=postgres dbname=registry sslmode=disable"
2024/07/03 09:21:11 Serving bulk issuance at http://[::]:5665`

SO please let me know is my container is running without error 
Or else please let me know any health check methods in order to ensure container is healthy like we check for http://url/credentials/health for credentials service  ....

**Reply by holashchand** [reply]
Hi @PrasadAthani , please follow the guide here, https://github.com/orgs/Sunbird-RC/discussions/1038#discussioncomment-9925150

**Reply by PrasadAthani** [reply]
Thanks a lot @holashchand
Now after copying that error Failed to find configuration ./application-default.yml is resolved 
But how to test this bulk issuance is service working in right manner or not?

**Reply by holashchand** [reply]
Hi @PrasadAthani, Nice to know you are able to fix this now. Checkout apis here and try out apis one by one. https://rc.sunbird.org/api-reference/bulk-issuance-api/get-sample-template
